src/test/java/org/testcontainers/cloud/turbomode/sample/AbstractIntegrationTest.java

```diff
@@ -0,0 +1,58 @@
+package org.testcontainers.cloud.turbomode.sample;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.awaitility.Duration;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class AbstractIntegrationTest {
+
+    static KafkaContainer kafka = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:6.2.1"));
+
+    static {
+        kafka.start();
+    }
+
+    private KafkaConsumer<String, String> createConsumer(String topic) {
+        Map<String, Object> kafkaConfig = new HashMap<>();
+        kafkaConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers());
+        kafkaConfig.put(ConsumerConfig.GROUP_ID_CONFIG, "tc-" + UUID.randomUUID());
+        kafkaConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        kafkaConfig.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                StringDeserializer.class.getName());
+        kafkaConfig.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                StringDeserializer.class.getName());
+
+        KafkaConsumer<String, String> consumer = new KafkaConsumer<>(kafkaConfig);
+        consumer.subscribe(Collections.singletonList(topic));
+        return consumer;
+    }
+
+    protected void writeAndVerify(String clientId, String topic) {
+        EventWriter notifier = new EventWriter(kafka.getBootstrapServers(), clientId, topic);
+        String notificationMessage = "foobar";
+        int messageCount = 100000;
+        for (int i = 0; i < messageCount; i++) {
+            notifier.sendNotification(notificationMessage);
+        }
+
+        // use consumer to check received records
+        KafkaConsumer<String, String> consumer = createConsumer(topic);
+
+        AtomicInteger readMessages = new AtomicInteger();
+        await().atMost(Duration.TWO_MINUTES).untilAsserted(() -> {
+            ConsumerRecords<String, String> records = consumer.poll(100);
+            readMessages.addAndGet(records.count());
+            assertEquals(readMessages.get(), messageCount);
```

Comment:
@kiview @clive-jevons the test were failing because we were writing in the same topic with another test, the `EventWriterJupiterTests` that we created, remember? After removing it, tests run in all 3 scenarios. So I think it's not needed to remove the already read messages [as here](https://github.com/AtomicJar/testcontainers-cloud-demo/blob/master/kafka-mongodb-repro/src/test/java/cloud/testcontainers/PugPartyInviteTest.java). WDYT?

Reply:
Oh right, no need for the `EventWriterJupiterTests`. In this case, fine with me 👍

Reply:
If we're not using "exactly once semantics" we run the risk of flaky tests if we don't use the "track UUID per message" approach. So we _could_ conceivably run into the case where for a user the tests fail - not because their setup is bad, but because they've run into the edge case where the message gets received twice.

But - it is also entirely legitimate to say that we'll wait until we've hit that problem before fixing it 😂 
I'll leave it up to you @ilopezluna